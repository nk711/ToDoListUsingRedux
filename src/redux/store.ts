import { configureStore } from "@reduxjs/toolkit";
import cardReducer from "./cardsSlice"

export const store = configureStore({
    reducer : {
        cards: cardReducer
    }
})

export type RootState  = ReturnType<typeof store.getState>;

export default store;