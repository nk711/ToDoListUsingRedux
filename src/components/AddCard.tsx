import React, { useState } from "react";
import { Alert, Button, StyleSheet, Text, TextInput, TouchableOpacity, View } from "react-native";
import { useDispatch } from "react-redux";
import { addCard } from "../redux/cardsSlice";

export const AddCard = () => {
    const [input, setInput] = useState('');
    //const setTodoList = useSetRecoilState(todoListState);

    const dispatch = useDispatch();


    const addItem = (e: any) => {
        e.preventDefault();
        console.log('adding card', input)
        dispatch(
            addCard({
                key: Math.random(),
                todo: input,
                isComplete: false
            })
        );
        setInput('')
    }

    const onChange = (val: string) => {
        setInput(val)
    }

    return (
        <View style = { styles.column }>
            <TextInput
                style = { styles.textInput}
                placeholder = 'Add something here you need to do!'
                value ={input} 
                multiline = { true }
                onChangeText={ text => onChange(text) }/>
            <TouchableOpacity 
                style = { styles.addButton }
                onPress= {addItem}>
                    <Text style = {styles.addButtonText}> + Add a card </Text>
            </TouchableOpacity>
        </View>
    )
}

const styles = StyleSheet.create({
    column : {
        maxWidth: '100%',
        flexDirection: 'column',
        justifyContent: 'space-between',
    },
    textInput: {
        width: '100%',
        backgroundColor: '#fff',
        borderRadius: 13,
        padding: 15,
        marginTop: 10,
        lineHeight: 25,
        fontSize: 20,
        fontWeight: '300',

    },
    addButton: {
        marginVertical: 10,
    },
    addButtonText: {
        fontWeight: '300',
        fontSize: 20,
    }
})

