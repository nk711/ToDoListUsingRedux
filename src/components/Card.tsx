import React from "react";
import { Button, StyleSheet, Text, TextInput, TouchableOpacity, View } from "react-native";
import { useDispatch, useSelector } from "react-redux";
import { CardItem, deleteCard } from "../redux/cardsSlice";


const replaceItemAtIndex = (arr: CardItem[], index: number, val: CardItem) => {
    return [...arr.slice(0, index), val, ...arr.slice(index+1)];
}

const removeItemAtIndex = (arr: CardItem[], index: number) => {
    return [...arr.slice(0, index), ...arr.slice(index+1)]
}

export const Card = ({item} : any) => { 
    const dispatch = useDispatch();
    const cardList = useSelector((state:any)=> {
        return state.cards
    })
    
    const index= cardList.findIndex((card: CardItem)=> card === item)

    const editItemText = (text: string) => {
        const newList = replaceItemAtIndex(cardList, index, {
            ...item,
            todo: text,
        });
        //setCardList(newList)
    }

    // const toggleComplete = () => {
    //     const newList = replaceItemAtIndex(cardList, index, {
    //         ...item,
    //         isComplete: !item.isComplete,
    //     });
    //     setCardList(newList);
    // };

    const deleteItem = () => {
        dispatch(
            deleteCard({
                key: item.key
            })
        )
    }

    return ( 
        <View style = {styles.container}>
            <TextInput
                style = {styles.textInput}
                value ={item.todo} 
                multiline = {true}
                // onChangeText={ text => editItemText(text) }
                />
            <View style={styles.row}>
                <TouchableOpacity
                    style = {styles.delete}
                    onPress= {deleteItem}>
                        <Text style = { styles.buttonText}>Delete</Text>
                </TouchableOpacity> 
                <TouchableOpacity
                    style = {
                        item.isComplete ? (styles.statusComplete) : (styles.statusInProgress)
                    }
                    // onPress= {toggleComplete}
                    >
                        <Text style = { styles.buttonText}>{ item.isComplete ? ('Is Complete') : ('In Progress')}</Text>
                </TouchableOpacity> 
            </View>
        </View>
    )

}


const styles = StyleSheet.create({
    container : {
        backgroundColor: '#fff',
        maxWidth: '100%',
        flexDirection: 'column',
        borderRadius: 13,
        marginVertical:10,
        paddingHorizontal: 10,
        paddingBottom: 10,
        justifyContent: 'space-between',
    },

    textInput: {
        width: '100%',
        color: '#000',
        fontWeight: '300',
        backgroundColor: '#fff',
        borderRadius: 13,
        padding: 15,
        lineHeight: 30,
        fontSize: 18,
    },
    delete: {
        borderRadius: 10,
        width: 70,
        height: 30,
        backgroundColor: '#b24b63',
        justifyContent: 'center',
        alignItems: 'center',
        marginRight: 10,

    },
    buttonText: {
        color:'#fff',
        fontSize: 15,
        fontWeight: '300',
    },
    statusComplete: {
        backgroundColor: '#99d372',
        borderRadius: 10,
        width: 100,
        height: 30,
        justifyContent: 'center',
        alignItems: 'center',
    },
    statusInProgress: {
        backgroundColor: 'orange',
        borderRadius: 10,
        width: 100,
        height: 30,
        justifyContent: 'center',
        alignItems: 'center',
    },
    row: {
        flexDirection: "row"
    }
})

