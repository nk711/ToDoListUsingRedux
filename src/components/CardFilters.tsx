import { Picker } from "@react-native-picker/picker"
import React from "react"
import { StyleSheet, Text } from "react-native"
import { useRecoilState } from "recoil"
import { todoListFilterState } from "../recoil/store"

export const CardFilters = () => {

    const [filter, setFilter] = useRecoilState(todoListFilterState)

    const updateFilter = (value: string) => {
        setFilter(value)
    }

    return ( 
        <>
        <Text style = {styles.options}> Filter By</Text>
        <Picker
            selectedValue={filter}
            onValueChange= {(value) => updateFilter(value)}>
            <Picker.Item style={styles.options} label='Show All' value='All'/>
            <Picker.Item style={styles.options} label='Show Completed' value='Completed'/>
            <Picker.Item style={styles.options} label='Show Uncompleted' value='Uncompleted'/>
        </Picker>
        </>
    )
}

const styles = StyleSheet.create({
    options: {
        fontSize: 18,
        fontWeight: '300'
    }
})

