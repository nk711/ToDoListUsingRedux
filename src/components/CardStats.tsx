import { Picker } from "@react-native-picker/picker"
import React from "react"
import { StyleSheet, Text, View } from "react-native"
import { useSelector } from "react-redux"
import { CardItem } from "../redux/cardsSlice"

export const CardStats = () => {
    
    const todoList = useSelector((state:any)=> {
        switch (state.cards.filter) {
            case 'Completed':
                return state.cards.items.filter((item: CardItem) => item.isComplete) 
            case 'InComplete':
                return state.cards.items.filter((item: CardItem) => !item.isComplete) 
            default:
                return state.cards.items
        }
    })
    
    const total = todoList.length;
    const totalCompleted = todoList.filter((item: CardItem) => item.isComplete).length;;
    const totalIncomplete = todoList.filter((item: CardItem) => !item.isComplete).length;
    const percentCompleted = total === 0 ? 0 : totalCompleted / total * 100;
    const formattedPercentCompleted = Math.round(percentCompleted)

    return ( 
        <View style={styles.stats}>
            <Text> * Total items: {total}</Text>
            <Text> * Items completed: {totalCompleted}</Text>
            <Text> * Items not completed: {totalIncomplete}</Text>
            <Text> * Percent completed: {formattedPercentCompleted}</Text>
        </View>
    )
}

const styles = StyleSheet.create({
    stats: {
        fontWeight: '600',
        fontSize: 30,
        paddingVertical: 20,
    }
})