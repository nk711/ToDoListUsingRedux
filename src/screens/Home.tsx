import React, { useEffect, useState } from "react";
import { Button, KeyboardAvoidingView, Platform, SafeAreaView, ScrollView, StatusBar, StyleSheet, Text, TextInput, View } from "react-native";
import { useSelector } from "react-redux";
import { AddCard } from "../components/AddCard";
import { Card } from "../components/Card";
import { CardFilters } from "../components/CardFilters";
import { CardStats } from "../components/CardStats";
import { CardItem } from "../redux/cardsSlice";

const randomInt = (min:number, max:number): number => { // min and max included 
    return Math.floor(Math.random() * (max - min + 1) + min)
}

  
export const Home = () => {
    const [bg, setBg] = useState('');
    const todoList = useSelector((state:any)=> {
        switch (state.cards.filter) {
            case 'Completed':
                return state.cards.items.filter((item: CardItem) => item.isComplete) 
            case 'InComplete':
                return state.cards.items.filter((item: CardItem) => !item.isComplete) 
            default:
                return state.cards.items
        }
    })

    const arrColors = [
        '#aacaaa', // green
        '#6f8493',
        '#d4a096',
        '#bfa440',
        '#c7cf6c',
        '#aacaaa',
        '#b24b63',
        '#4d77e7',
        '#b55799',
        '#aacaaa',
        '#4ab6c2',
        '#aacaaa',
        '#cbb086',
        '#99d372',
        '#aacaaa',
    ]
    
    useEffect( () => {
        setBg(arrColors[randomInt(0,arrColors.length)])
    }, [todoList])

    return (<>
            <SafeAreaView>
                <StatusBar translucent backgroundColor= {bg}/>
            </SafeAreaView>
            <KeyboardAvoidingView
                behavior= {Platform.OS === 'ios' ? 'padding' : 'height'} 
            >
                <ScrollView testID = "Home.ScrollView" style= {[styles.background, {backgroundColor: bg}]}>
                    <Text style = {styles.title}> To Do List </Text>
                    <View style = {styles.cardList}>
                        <CardFilters />
                        <AddCard/>
                        { todoList.map((item: any)=> (
                            <Card key = {item.key} item = {item}>
                                {item}
                            </Card>
                        ))}
                    </View>
                       <CardStats/>
                </ScrollView>
            </KeyboardAvoidingView>
            </>
)
}

const styles = StyleSheet.create({
    background: {
        height: '100%',
        padding: 20,
    },
    row: {
        flexDirection: 'row'
    },
    filter: {
        width: '50%'
    },
    cardList: {
        borderRadius: 15,
        padding: 25,
        backgroundColor: '#ebecf0',
        paddingBottom: 50,
    },
    title: {
        fontWeight: '600',
        fontSize: 30,
        paddingVertical: 20,
    }
})


