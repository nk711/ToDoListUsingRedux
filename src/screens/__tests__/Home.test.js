import 'react-native';
import React from 'react';
import { Provider as StoreProvider } from 'react-redux';
// Note: test renderer must be required after react-native.
import {render, fireEvent} from '@testing-library/react-native';
import { Home } from '../Home';
import { Card } from '../../components/Card'
import { AddCard } from '../../components/AddCard';
import store from '../../redux/store'
import { useEffect } from 'react';
import { configureStore } from '@reduxjs/toolkit';

const component = (
  <StoreProvider store = {store}>
    <Home/>
  </StoreProvider>
)

const percentCompleted = (completed, total) => {
  return Math.round(total === 0 ? 0 :  completed / total * 100);
}

describe('Check Home Screen Components', () => {
  test('Given the user goes to the Home Screen, the screen should render as expected', () => {
    jest.spyOn(global.Math, 'floor').mockReturnValue(0);
    const {toJSON } = render(component);
    expect(toJSON()).toMatchSnapshot();
  });

  test('Given the user is on the home screen, the add card input should be rendered', () => {
    const {getByTestId } = render(component);
    const inputField = getByTestId('AddCard-Input');
    expect(inputField).toBeTruthy();
  });


});

describe('Testing Card Component', () => {
  test('should be able to add a new card', () => {
    const {toJSON, getByTestId} = render(component);
    const inputField = getByTestId('AddCard-Input');
    const addCardButton = getByTestId('AddCard-Button');
    fireEvent.changeText(inputField, 'Test Message');
    fireEvent.press(addCardButton)
    const card =  store.getState().cards.items[0]

    const item = {
      key: 1,
      todo: 'Test Message',
      isComplete: false,
    }

    expect(card).toEqual(item)
  });

  test('should display card todo input text field', () => {
    const {getByTestId } = render(component);
    const editInputField = getByTestId('Card-Input-1');
    expect(editInputField).toBeTruthy();
  });

  test("Given that the Card input field has been edited with the message 'Edited Message',- the card redux state should be updated", () => {
    const {toJSON, getByTestId} = render(component);    
    const editInputField = getByTestId('Card-Input-1');
    fireEvent.changeText(editInputField, 'Edited Message');

    const card =  store.getState().cards.items[0]
    expect(card.todo).toEqual('Edited Message')
  });

  test('should be able to set a card to completed', () => {
    const {toJSON, getByTestId} = render(component);    
    const isCompleteButton = getByTestId('Card-IsComplete-1');
    fireEvent.press(isCompleteButton, 'Edited Message');
    const card =  store.getState().cards.items[0]
    expect(card.isComplete).toBe(true)
  });

  test('should be able to delete a card', () => {
    const {toJSON, getByTestId} = render(component);    
    const deleteButton = getByTestId('Card-Delete-1');
    fireEvent.press(deleteButton, 'Edited Message');
    const list =  store.getState().cards.items
    expect(list).toHaveLength(0)
  });

});



describe('Testing if stats render correctly', () => {
  
  beforeAll (() => {
    // Adding list of cards
    const {toJSON, getByTestId} = render(component);
    const inputField = getByTestId('AddCard-Input');
    const addCardButton = getByTestId('AddCard-Button');
    for (let i= 0; i<10; i++) {
      fireEvent.changeText(inputField, 'Test Message ' + i);
      fireEvent.press(addCardButton);
    }
  });

  test('Check if Completed Items label is rendered ', () => {
    const {toJSON, getByTestId} = render(component);    
    const completedItems = getByTestId('CompletedItems');
    expect(completedItems).toBeTruthy();
  });

  test('Check if Completed Items is set as 0 items', () => {
    const {toJSON, getByTestId} = render(component);    
    const completedItems = getByTestId('CompletedItems');
    expect(completedItems.props.children).toEqual([" * Items completed: ", 0]);
  });

  test('Check if Total Items label is rendered ', () => {
    const {toJSON, getByTestId} = render(component);    
    const totalItems = getByTestId('TotalItems');
    expect(totalItems).toBeTruthy();
  });

  test('Check if Total Items label is set as 10 items', () => {
    const {toJSON, getByTestId} = render(component);    
    const totalItems = getByTestId('TotalItems');
    expect(totalItems.props.children).toEqual([" * Total items: ", 10]);
  });

  test('Check if Incomplete Items label is rendered ', () => {
    const {toJSON, getByTestId} = render(component);    
    const incompleteItems = getByTestId('IncompleteItems');
    expect(incompleteItems).toBeTruthy();
  });

  test('Check if Incomplete Items label is set as 10 ', () => {
    const {toJSON, getByTestId} = render(component);    
    const incompleteItems = getByTestId('IncompleteItems');
   expect(incompleteItems.props.children).toEqual([" * Items not completed: ", 10]);
  });

  test('Check if percentage label is rendered ', () => {
    const {toJSON, getByTestId} = render(component);    
    const percentageText = getByTestId('Percentage');
    expect(percentageText).toBeTruthy();
  });

  test('Check if percentage label is set as 0', () => {
    const {toJSON, getByTestId} = render(component);    
    const percentageText = getByTestId('Percentage');
    expect(percentageText.props.children).toEqual([" * Percent completed: ", 0]);

  });

  test('Check if Completed Items is set as 5 items after setting 5 cards as completed', () => {
    const {toJSON, getByTestId} = render(component);    
    for (let i= 1; i<6; i++) {
      let isCompleteButton = getByTestId('Card-IsComplete-'+i);
      fireEvent.press(isCompleteButton);
    }
    const completedItems = getByTestId('CompletedItems');
    expect(completedItems.props.children).toEqual([" * Items completed: ", 5]);
  });

  test('Check if Total Items label is set as 10 items after setting 5 cards as completed', () => {
    const {toJSON, getByTestId} = render(component);    
    const totalItems = getByTestId('TotalItems');
    expect(totalItems.props.children).toEqual([" * Total items: ", 10]);

  });

  
  test('Check if Incomplete Items label is set as 5 items after setting 5 cards as completed', () => {
    const {toJSON, getByTestId} = render(component);    
    const incompleteItems = getByTestId('IncompleteItems');
    expect(incompleteItems.props.children).toEqual([" * Items not completed: ", 5]);
  });


  test('Check if percentage label is set as 50, after setting 5 cards as completed', () => {
    const {toJSON, getByTestId} = render(component);    
    const percentageText = getByTestId('Percentage');
    expect(percentageText.props.children).toEqual([" * Percent completed: ", 50]);

  });




});


describe('Test Card Filter Component', () => {

  beforeAll (() => {
    // Adding list of cards
    const {toJSON, getByTestId} = render(component);
    const inputField = getByTestId('AddCard-Input');
    const addCardButton = getByTestId('AddCard-Button');
    let i = 0;
    for (i= 1; i<=10; i++) {
      fireEvent.changeText(inputField, 'Test Message ' + i);
      fireEvent.press(addCardButton);
    }

    let isCompleteButton;
    for (i= 1; i<4; i++) {
      isCompleteButton = getByTestId('Card-IsComplete-'+i);
      fireEvent.press(isCompleteButton);
    }
  });

  test('Check if filter value can be changed to Completed', () => {
    const { getByTestId} = render(component);    
    const filterInput = getByTestId('FilterInput');
    fireEvent(filterInput, 'valueChange', 'Completed')
    expect(store.getState().cards.filter).toBe('Completed')
  });

  test('Check if filter value can be changed to Incomplete', () => {
    const {getByTestId} = render(component);    
    const filterInput = getByTestId('FilterInput');
    fireEvent(filterInput, 'valueChange', 'Incomplete')
    expect(store.getState().cards.filter).toBe('Incomplete')
  });

  test('Check if filter value can be changed to All', () => {
    const { getByTestId} = render(component);    
    const filterInput = getByTestId('FilterInput');
    fireEvent(filterInput, 'valueChange', 'All')
    expect(store.getState().cards.filter).toBe('All')
  });

  test('Check if changing filter value changes', () => {
    const { getByTestId} = render(component);    
    const filterInput = getByTestId('FilterInput');
    fireEvent(filterInput, 'valueChange', 'Completed')
    expect(store.getState().cards.filter).toBe('Completed')
  });

  test('Check if cards get filtered when set to Completed', () => {
    const {toJSON, getByTestId} = render(component);    
    const filterInput = getByTestId('FilterInput');
    fireEvent(filterInput, 'valueChange', 'Completed')
    expect(toJSON()).toMatchSnapshot();
  });

  test('Check if cards get filtered when set to All', () => {
    const {toJSON, getByTestId} = render(component);    
    const filterInput = getByTestId('FilterInput');
    fireEvent(filterInput, 'valueChange', 'All')
    expect(toJSON()).toMatchSnapshot();
  });

  test('Check if cards get filtered when set to Incomplete', () => {
    const {toJSON, getByTestId} = render(component);    
    const filterInput = getByTestId('FilterInput');
    fireEvent(filterInput, 'valueChange', 'Incomplete')
    expect(toJSON()).toMatchSnapshot();
  });


})

