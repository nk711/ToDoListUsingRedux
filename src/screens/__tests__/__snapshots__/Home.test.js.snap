// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding a card user can add a card and the redux state will be updated with the new card. 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  />,
  <View
    onLayout={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "height": "100%",
            "padding": 20,
          },
          Object {
            "backgroundColor": "#aacaaa",
          },
        ]
      }
      testID="Home.ScrollView"
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "600",
              "paddingVertical": 20,
            }
          }
        >
           To Do List 
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ebecf0",
              "borderRadius": 15,
              "padding": 25,
              "paddingBottom": 50,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "300",
              }
            }
          >
             Filter By
          </Text>
          <View>
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "Show All",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "All",
                  },
                  Object {
                    "label": "Show Completed",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "Completed",
                  },
                  Object {
                    "label": "Show Incomplete",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "Incomplete",
                  },
                ]
              }
              numberOfLines={1}
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "space-between",
                "maxWidth": "100%",
              }
            }
          >
            <TextInput
              multiline={true}
              onChangeText={[Function]}
              placeholder="Add something here you need to do!"
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 13,
                  "fontSize": 20,
                  "fontWeight": "300",
                  "lineHeight": 25,
                  "marginTop": 10,
                  "padding": 15,
                  "width": "100%",
                }
              }
              testID="AddCard.Input"
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginVertical": 10,
                  "opacity": 1,
                }
              }
              testID="AddCard.Button"
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "fontWeight": "300",
                  }
                }
              >
                 + Add a card 
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 13,
                "flexDirection": "column",
                "justifyContent": "space-between",
                "marginVertical": 10,
                "maxWidth": "100%",
                "paddingBottom": 10,
                "paddingHorizontal": 10,
              }
            }
          >
            <TextInput
              multiline={true}
              onChangeText={[Function]}
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 13,
                  "color": "#000",
                  "fontSize": 18,
                  "fontWeight": "300",
                  "lineHeight": 30,
                  "padding": 15,
                  "width": "100%",
                }
              }
              testID="Card.Input"
              value="Test Message"
            />
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#b24b63",
                    "borderRadius": 10,
                    "height": 30,
                    "justifyContent": "center",
                    "marginRight": 10,
                    "opacity": 1,
                    "width": 70,
                  }
                }
                testID="Card.Delete"
              >
                <Text
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 15,
                      "fontWeight": "300",
                    }
                  }
                >
                  Delete
                </Text>
              </View>
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "orange",
                    "borderRadius": 10,
                    "height": 30,
                    "justifyContent": "center",
                    "opacity": 1,
                    "width": 100,
                  }
                }
                testID="Card.IsComplete"
              >
                <Text
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 15,
                      "fontWeight": "300",
                    }
                  }
                >
                  In Progress
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "600",
              "paddingVertical": 20,
            }
          }
        >
          <Text>
             * Total items: 
            1
          </Text>
          <Text>
             * Items completed: 
            0
          </Text>
          <Text>
             * Items not completed: 
            1
          </Text>
          <Text>
             * Percent completed: 
            0
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>,
]
`;

exports[`Check Home Screen Components renders as expected 1`] = `
Array [
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  />,
  <View
    onLayout={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <RCTScrollView
      style={
        Array [
          Object {
            "height": "100%",
            "padding": 20,
          },
          Object {
            "backgroundColor": "#aacaaa",
          },
        ]
      }
      testID="Home.ScrollView"
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "600",
              "paddingVertical": 20,
            }
          }
        >
           To Do List 
        </Text>
        <View
          style={
            Object {
              "backgroundColor": "#ebecf0",
              "borderRadius": 15,
              "padding": 25,
              "paddingBottom": 50,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "300",
              }
            }
          >
             Filter By
          </Text>
          <View>
            <RNCPicker
              items={
                Array [
                  Object {
                    "label": "Show All",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "All",
                  },
                  Object {
                    "label": "Show Completed",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "Completed",
                  },
                  Object {
                    "label": "Show Incomplete",
                    "testID": undefined,
                    "textColor": undefined,
                    "value": "Incomplete",
                  },
                ]
              }
              numberOfLines={1}
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
          <View
            style={
              Object {
                "flexDirection": "column",
                "justifyContent": "space-between",
                "maxWidth": "100%",
              }
            }
          >
            <TextInput
              multiline={true}
              onChangeText={[Function]}
              placeholder="Add something here you need to do!"
              style={
                Object {
                  "backgroundColor": "#fff",
                  "borderRadius": 13,
                  "fontSize": 20,
                  "fontWeight": "300",
                  "lineHeight": 25,
                  "marginTop": 10,
                  "padding": 15,
                  "width": "100%",
                }
              }
              testID="AddCard.Input"
              value=""
            />
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginVertical": 10,
                  "opacity": 1,
                }
              }
              testID="AddCard.Button"
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                    "fontWeight": "300",
                  }
                }
              >
                 + Add a card 
              </Text>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "fontSize": 30,
              "fontWeight": "600",
              "paddingVertical": 20,
            }
          }
        >
          <Text>
             * Total items: 
            0
          </Text>
          <Text>
             * Items completed: 
            0
          </Text>
          <Text>
             * Items not completed: 
            0
          </Text>
          <Text>
             * Percent completed: 
            0
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>,
]
`;
